koku/api/provider/test/tests_views.py

```diff
@@ -204,3 +211,161 @@ def test_create_provider_as_service_admin(self):
+        other_user = self.create_user(other_owner_token, other_user_data)
+        self.assertEquals(other_user.status_code, 201)
+        other_user_token = self.get_token(other_user_data['username'], other_user_data['password'])
+
+        # Remove Provider as the regular user that belongs to the other company
+        url = reverse('provider-detail', args=[json_result.get('uuid')])
+        client = APIClient()
+        client.credentials(HTTP_AUTHORIZATION=other_user_token)
+        response = client.delete(url)
+        self.assertEqual(response.status_code, 403)
+
+    def test_remove_provider_with_reg_user_same_company_didnt_create(self):
+        """
+        Test removing a provider by a regular user of the same company but did not create the provider.
+
+        PermissionDenied is expected.
+        """
+        # Get the customer owner token so a regular user can be created
+        user_name = self.customer_data[0]['owner']['username']
+        user_pass = self.customer_data[0]['owner']['password']
+        owner_token = self.get_token(user_name, user_pass)
+
+        # Create a regular user and get the user's token
+        user_data = {'username': 'james', 'password': 'jamespassword', 'email': 'james@me.com'}
+        user = self.create_user(owner_token, user_data)
+        self.assertEquals(user.status_code, 201)
+        user_token = self.get_token(user_data['username'], user_data['password'])
+
+        # Create a Provider as a regular user
+        iam_arn = 'arn:aws:s3:::my_s3_bucket'
+        bucket_name = 'my_s3_bucket'
+        response = self.create_provider(bucket_name, iam_arn, user_token)
+        self.assertEqual(response.status_code, 201)
+
+        # Verify that the Provider creation was successful
+        json_result = response.json()
+        self.assertIsNotNone(json_result.get('uuid'))
+        self.assertIsNotNone(json_result.get('customer'))
+        self.assertEqual(json_result.get('customer').get('name'),
+                         self.customer_data[0].get('name'))
+        self.assertIsNotNone(json_result.get('created_by'))
+        self.assertEqual(json_result.get('created_by').get('username'), user_data['username'])
+
+        # Create another regular user and get the user's token
+        other_user_data = {'username': 'sally', 'password': 'sallypassword', 'email': 'sally@me.com'}
+        other_user = self.create_user(owner_token, other_user_data)
+        self.assertEquals(other_user.status_code, 201)
+        other_user_token = self.get_token(other_user_data['username'], other_user_data['password'])
+
+        # Remove Provider as a regular user that belongs to the same company but did not create the Provider
+        url = reverse('provider-detail', args=[json_result.get('uuid')])
+        client = APIClient()
+        client.credentials(HTTP_AUTHORIZATION=other_user_token)
+        response = client.delete(url)
```

Comment:
Think this needs an assertion here?

Reply:
yup, before it was just passively verifying that nothing blew up.